room-distances/calculate-distances.py

```diff
@@ -0,0 +1,133 @@
+import requests
+import json
+from pathlib import Path
+from math import ceil
+from geopy import distance
+import re
+
+
+MAP_URL = 'https://assets.neuland.app/rooms_neuland.geojson'
+ROOM_TYPES = ['Hörsaal', 'PC-Pool', 'Vorlesung', 'Seminarraum', 'Labor']
+STAIRCASE_TYPES = ['Treppenhaus']
+
+
+# calculate center of room from coordinates
+def calculate_center(room):
+    lat = []
+    long = []
+
+    for l in room['geometry']['coordinates'][0]:
+        lat.append(l[0])
+        long.append(l[1])
+
+    return [sum(lat)/len(lat), sum(long)/len(long)]
+
+
+def findNearestStaircase(room, staircases):
```

Comment:
when i commented about incorperating floor levels i just thought about adding e.g. 10m for each floor. I did not realise incorperating moving to staircases and back would make this problem this complicated.
Your solution looks pretty good though.

Does this code work for paths from/to building A? I think the staircases in A which are floating and sometimes not connecting all floors are not mapped correctly or just missing?

Reply:
Just adding 10m per floor was also my first approach, however I think the results are then a bit inaccurate in terms of sorting for room suggestions. 

However, the current implementation is also somewhat naive, as it assumes that every staircase reaches every floor. Accordingly, the results in building A could be rather moderate, but sufficient for accurate sorting. Of course, since this is not real navigation, this implementation should be sufficient.